src/config.rs

```diff
@@ -8,27 +8,31 @@ use std::fs;
 use std::path::PathBuf;
 
 pub const DB_MAX_SIZE: u64 = 8 * 1024 * 1024; // 8 MiB
-
 pub const DB_VERSION: DBVersion = 3;
 
-pub fn zo_data() -> Result<PathBuf> {
-    let path = match env::var_os("_ZO_DATA") {
-        Some(data_osstr) => PathBuf::from(data_osstr),
+pub fn zo_data_dir() -> Result<PathBuf> {
+    match env::var_os("_ZO_DATA_DIR") {
+        Some(data_osstr) => {
+            let data_dir = PathBuf::from(data_osstr);
+
+            if data_dir.is_file() {
+                bail!("_ZO_DATA_DIR should point to a directory, not a file");
+            }
+
+            Ok(data_dir)
+        }
         None => {
-            if let Some(mut cache_dir) = dirs::cache_dir() {
-                cache_dir.push("zoxide");
-                cache_dir
-            } else if let Some(mut home_dir) = dirs::home_dir() {
-                home_dir.push(".zoxide");
-                home_dir
+            if let Some(xdg_data_dir) = dirs::data_dir() {
```

Comment:
Use `dirs::data_local_dir()` instead. On Windows, `dirs::data_dir()` defaults to `C:\Users\Alice\AppData\Roaming`, which is meant to be shared by the same user across different computers.